packages/composite-checkout-wpcom/src/components/coupon.js

```diff
@@ -108,13 +111,18 @@ function handleFormSubmit(
 	couponFieldValue,
 	setHasCouponError,
 	couponAdded,
-	setIsCouponApplied
+	setIsCouponApplied,
+	onEvent
 ) {
 	event.preventDefault();
 
 	//TODO: Validate coupon field and replace condition in the following if statement
 	if ( couponFieldValue === 'Add' ) {
 		setIsCouponApplied( true );
+		onEvent( {
+			type: 'a8c_checkout_add_coupon',
+			payload: { coupon: couponFieldValue, client_id: 'WordPress.com' },
```

Comment:
Should we remove this client ID here since we can add these on the host?

Reply:
Yeah, it's probably not necessary, but I guess it could be used to differentiate between events in the composite-checkout package vs. events from calypso (like this one), if there's some reason we need to know that.

Reply:
I'll take it out for now, we can add them back later if they're necessary.